### ISSUE TITLE
Input from variable

### ISSUE THREAD

**Issue by thorst** [alafr/SVG-to-PDFKit#28]
Ideally I wouldn't have to write out to the DOM before importing the svg. If I have to, then I have to, but I wanted to attempt not doing so. Below is my attempts at doing so and ultimately using the same method as your examples to get it working. My question: **Is there a way to get Method 1 working?**

The core html is the same as that (or thereabouts) of ticket #26. 

There are two potentials for output. 1) The black boxes. 2) The correct output.
![image](https://cloud.githubusercontent.com/assets/1334950/24610731/baeb86fc-184d-11e7-9868-d416e8a70280.png)
![image](https://cloud.githubusercontent.com/assets/1334950/24610749/c99f8194-184d-11e7-9777-ef3e938ebe5a.png)

  
```
// Method #1 - Pure HTML, true or false show black boxes
//SVGtoPDF(doc, app.mermaid.html, doc.x, doc.y, { useCSS: true });

// Method #2 - jQuery in memory dom object, true or false show black boxes
//SVGtoPDF(doc, $.parseHTML(app.mermaid.html)[0], doc.x, doc.y, { useCSS: true });

// Method #3 - jQuery in memory dom object, true or false show black boxes
//let hiddenDiv = $("<svg>").attr({ 'xmlns': "http://www.w3.org/2000/svg", 'xmlns:xlink': "http://www.w3.org/1999/xlink" }).html(app.mermaid.html);
//SVGtoPDF(doc, hiddenDiv[0].firstChild, doc.x, doc.y, { useCSS: true });

// Method #4 - jQuery in memory dom object, true or false show black boxes
//SVGtoPDF(doc, $.parseHTML('<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">' + app.mermaid.html + '</svg>')[0].firstChild, doc.x, doc.y, { useCSS: true });

// Method #5 - hidden DOM object, true WORKS, false show black boxes
let hiddenDiv = $("#hidden-div").html('<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">' + app.mermaid.html + '</svg>');
SVGtoPDF(doc, hiddenDiv[0].firstChild, doc.x, doc.y, { useCSS: true });
```

**Comment by alafr** [comment]
Unfortunately only method `#5` works when there is use of CSS styling. It would be too complicated to parse css stylesheets and calculate precedence & inheritance.

**Comment by thorst** [comment]
No problem, this can be closed. I just wanted to make sure I was doing it the most efficient way possible.

**Comment by thorst** [comment]
Ugh. Now I'm getting the black boxes in IE 11. Have you seen this work in IE 11?

**Comment by thorst** [comment]
Do you want me to do a new issue?

Ok, To reproduce go to https://alafr.github.io/SVG-to-PDFKit/examples/demo.htm paste:
`<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"><svg id="mermaidId0" width="583" xmlns="http://www.w3.org/2000/svg" height="146" viewBox="0 0 603 166"><style type="text/css" title="mermaid-svg-internal-css">/*  */
#mermaidId0 .node&gt;rect { ; }
#mermaidId0 .node text  { fill:#000; stroke:none; font-weight:300; font-family:"Helvetica Neue",Helvetica,Arial,sans-serf; font-size:14px; }
#mermaidId0 .edgeLabel text  { fill:#000; stroke:none; font-weight:300; font-family:"Helvetica Neue",Helvetica,Arial,sans-serf; font-size:14px; }
#mermaidId0 .cluster rect  { rx:4px; fill: rgb(255, 255, 222); rx: 4px; stroke: rgb(170, 170, 51); stroke-width: 1px; }
*, ::before, ::after { box-sizing: border-box;}
.label { display: inline; padding: 0.2em 0.6em 0.3em; font-size: 75%; font-weight: bold; line-height: 1; color: rgb(255, 255, 255); text-align: center; white-space: nowrap; vertical-align: baseline; border-radius: 0.25em;}
.node rect, .node circle, .node ellipse, .node polygon { fill: rgb(236, 236, 255); stroke: rgb(204, 204, 255); stroke-width: 1px;}
.edgePath .path { stroke: rgb(51, 51, 51);}
.edgeLabel { background-color: rgb(232, 232, 232);}
.cluster rect { fill: rgb(255, 255, 222) !important; rx: 4 !important; stroke: rgb(170, 170, 51) !important; stroke-width: 1px !important;}
.cluster text { fill: rgb(51, 51, 51);}
.node text { font-family: "trebuchet ms", verdana, arial; font-size: 14px;}
* { padding: 0px; margin: 0px;}
/*  */
</style><g><g class="output"><g class="clusters"><g class="cluster" id="subGraph0" style="opacity: 1;" transform="translate(387,73)"><rect width="352" height="106" x="-176" y="-53"></rect><g class="label"><g transform="translate(0,0)"><text></text></g></g><text x="0" y="-39" fill="black" stroke="none" id="mermaidId0Text" style="text-anchor: middle;"> Cloverleaf</text></g></g><g class="edgePaths"><g class="edgePath" style="opacity: 1;"><path class="path" d="M342,73L367,73L392,73" marker-end="url(#arrowhead29)" style="stroke: #333; stroke-width: 1.5px;fill:none"></path><defs><marker id="arrowhead29" viewBox="0 0 10 10" refX="9" refY="5" markerUnits="strokeWidth" markerWidth="8" markerHeight="6" orient="auto"><path d="M 0 0 L 10 5 L 0 10 z" class="arrowheadPath" style="stroke-width: 1; stroke-dasharray: 1, 0;"></path></marker></defs></g><g class="edgePath" style="opacity: 1;"><path class="path" d="M161,73L186,73L211,73L236,73" marker-end="url(#arrowhead30)" style="stroke: #333; stroke-width: 1.5px;fill:none"></path><defs><marker id="arrowhead30" viewBox="0 0 10 10" refX="9" refY="5" markerUnits="strokeWidth" markerWidth="8" markerHeight="6" orient="auto"><path d="M 0 0 L 10 5 L 0 10 z" class="arrowheadPath" style="stroke-width: 1; stroke-dasharray: 1, 0;"></path></marker></defs></g></g><g class="edgeLabels"><g class="edgeLabel" style="opacity: 1;" transform=""><g transform="translate(0,0)" class="label"><rect rx="0" ry="0" width="0" height="0" style="fill:#e8e8e8;"></rect><text><tspan xml:space="preserve" dy="1em" x="1"></tspan></text></g></g><g class="edgeLabel" style="opacity: 1;" transform=""><g transform="translate(0,0)" class="label"><rect rx="0" ry="0" width="0" height="0" style="fill:#e8e8e8;"></rect><text><tspan xml:space="preserve" dy="1em" x="1"></tspan></text></g></g></g><g class="nodes"><g class="node" style="opacity: 1;" id="mainthread" transform="translate(289,73)"><rect rx="0" ry="0" x="-53" y="-18" width="106" height="36" style="fill:#cde498;stroke:#13540c;"></rect><g class="label" transform="translate(0,0)"><g transform="translate(-43,-8)"><text><tspan xml:space="preserve" dy="1em" x="1">270_FR_EPIC</tspan></text></g></g></g><g class="node" style="opacity: 1;" id="270_TO_PASSPORT" transform="translate(465,73)"><rect rx="0" ry="0" x="-73" y="-18" width="146" height="36"></rect><g class="label" transform="translate(0,0)"><g transform="translate(-63,-8)"><text><tspan xml:space="preserve" dy="1em" x="1">270_TO_PASSPORT</tspan></text></g></g></g><g class="node" style="opacity: 1;" id="protocol" transform="translate(90.5,73)"><rect rx="0" ry="0" x="-70.5" y="-25" width="141" height="50"></rect><g class="label" transform="translate(0,0)"><g transform="translate(-60.5,-15)"><text><tspan xml:space="preserve" dy="1em" x="1">Type:TcpIp/server</tspan><tspan xml:space="preserve" dy="1em" x="1">Connection:17981</tspan></text></g></g></g></g></g></g></svg></svg>`

And check the css box.

In chrome this works fine, and in IE 11 it prompts to save the file which is fine, but then you open in adobe and it indeed is black boxes

**Comment by alafr** [comment]
No need to open a new issue.
It should be fixed by commit b488240, at least it is in my IE11 installation
Thanks for reporting this bug!

**Comment by thorst** [comment]
Yes, That fixes it. THANKS! This project has been a life-saver.
